coherence-distributed-configuration/src/main/java/io/micronaut/coherence/discovery/CoherenceConfigurationClient.java

```diff
@@ -0,0 +1,182 @@
+    private static final String DEFAULT_APPLICATION = "application";
+    /**
+     * The name of the map that is used to store configuration properties.
+     */
+    private static final String CONFIG_CACHE_NAME = "config-cache";
+
+    private final ApplicationConfiguration applicationConfiguration;
+    private final List<Flowable<PropertySource>> propertySources = new ArrayList<>();
+    private final CoherenceClientConfiguration coherenceClientConfiguration;
+
+    public CoherenceConfigurationClient(ApplicationConfiguration applicationConfiguration,
+                                        CoherenceClientConfiguration coherenceClientConfiguration) {
+        this.applicationConfiguration = applicationConfiguration;
+        this.coherenceClientConfiguration = coherenceClientConfiguration;
+        LOG.debug("Coherence Client configuration: " + coherenceClientConfiguration);
+    }
+
+    @Override
+    public Publisher<PropertySource> getPropertySources(Environment environment) {
+        ValueExtractor<String, String> keyExtractor = new KeyExtractor<>();
+        NamedCache<String, Object> configCache = getCache(CONFIG_CACHE_NAME, coherenceClientConfiguration);
+
+        Map<Integer, String> keys = buildKeys(applicationConfiguration, environment);
+        for (Map.Entry<Integer, String> entry : keys.entrySet()) {
+            final Integer priority = entry.getKey();
+            final String propertySource = entry.getValue();
+            final String propertySourcePath = propertySource + "/";
+            Remote.Predicate<String> pathPredicate = cacheKey -> cacheKey.startsWith(propertySourcePath) &&
+                    cacheKey.length() > propertySourcePath.length() &&
+                    !cacheKey.substring(propertySourcePath.length()).contains("/");
+            Filter<String> filter = Filters.predicate(keyExtractor, pathPredicate);
+            Map<String, Object> props = configCache.entrySet(filter)
+                    .stream()
+                    .collect(Collectors.toMap((Converter<Map.Entry<String, Object>, String>)
+                                    cacheEntry -> cacheEntry.getKey().substring(propertySourcePath.length()),
+                            Map.Entry::getValue));
+            Flowable<PropertySource> propertySourceFlowable = Flowable.just(PropertySource.of(propertySource, props, priority));
+            propertySources.add(propertySourceFlowable);
+        }
+        return Flowable.merge(propertySources);
+    }
+
+    /**
+     * Returns NamedCache instance.
+     *
+     * @param cacheName                    cache name
+     * @param coherenceClientConfiguration configuration
+     * @return cache
+     */
+    protected NamedCache<String, Object> getCache(String cacheName, CoherenceClientConfiguration coherenceClientConfiguration) {
+        Session session = buildSession(coherenceClientConfiguration);
+        return session.getCache(cacheName);
```

Comment:
Hmm... We need to make sure we get the session for the `$SYS` scope, and get the cache from it.

Check with @thegridman on how to do that when using gRPC client.